libs/langchain-google-common/src/output_parsers.ts

```diff
@@ -0,0 +1,261 @@
+      .reduce(
+        (previousValue, currentValue) => `${previousValue}${currentValue}`
+      );
+  }
+
+  abstract segmentPrefix(
+    grounding: GroundingInfo,
+    support: GeminiGroundingSupport,
+    index: number
+  ): string | undefined;
+
+  abstract segmentSuffix(
+    grounding: GroundingInfo,
+    support: GeminiGroundingSupport,
+    index: number
+  ): string | undefined;
+
+  annotateSegment(
+    text: string,
+    grounding: GroundingInfo,
+    support: GeminiGroundingSupport,
+    index: number
+  ): string {
+    const start = support.segment.startIndex ?? 0;
+    const end = support.segment.endIndex;
+
+    const textBefore = text.substring(0, start);
+    const textSegment = text.substring(start, end);
+    const textAfter = text.substring(end);
+
+    const textPrefix = this.segmentPrefix(grounding, support, index) ?? "";
+    const textSuffix = this.segmentSuffix(grounding, support, index) ?? "";
+
+    return `${textBefore}${textPrefix}${textSegment}${textSuffix}${textAfter}`;
+  }
+
+  annotateTextSegments(text: string, grounding: GroundingInfo): string {
+    // Go through each support info in reverse, since the segment info
+    // is sorted, and we won't need to adjust string indexes this way.
+    let ret = text;
+    for (let co = grounding.supports.length - 1; co >= 0; co -= 1) {
+      const support = grounding.supports[co];
+      ret = this.annotateSegment(ret, grounding, support, co);
+    }
+    return ret;
+  }
+
+  abstract textPrefix(
+    text: string,
+    grounding: GroundingInfo
+  ): string | undefined;
+
+  abstract textSuffix(
+    text: string,
+    grounding: GroundingInfo
+  ): string | undefined;
+
+  /**
+   * Google requires us to
+   * "Display the Search Suggestion exactly as provided without any modifications"
+   * So this will typically be called from the textSuffix() method to get
+   * a string that renders HTML.
+   * See https://ai.google.dev/gemini-api/docs/grounding/search-suggestions
+   * @param grounding
+   */
+  searchSuggestion(grounding: GroundingInfo): string {
+    return grounding.metadata.searchEntryPoint?.renderedContent ?? "";
+  }
+
+  annotateText(text: string, grounding: GroundingInfo): string {
+    const prefix = this.textPrefix(text, grounding) ?? "";
+    const suffix = this.textSuffix(text, grounding) ?? "";
+    const body = this.annotateTextSegments(text, grounding);
+    return `${prefix}${body}${suffix}`;
+  }
+
+  parseResult(
+    generations: Generations,
+    _callbacks?: Callbacks
+  ): Promise<string> {
+    const text = this.generationsToString(generations);
+
+    const grounding = this.generationsToGroundingInfo(generations);
+    if (!grounding) {
+      return Promise.resolve(text);
+    }
+
+    return Promise.resolve(this.annotateText(text, grounding));
+  }
+}
+
+export class SimpleGoogleSearchOutputParser extends BaseGoogleSearchOutputParser {
+  // FIXME: What should this be?
+  lc_namespace: string[] = ["google_common", "output_parsers"];
```

Comment:
This is fine, won't serialize these anyway

Just put it on the base abstract class